README.md

```diff
@@ -0,0 +1,1075 @@
+        value should not be included in the result.
+      </td>
+    </tr>
+    <tr>
+      <td>done</td>
+      <td>function</td>
+      <td>null</td>
+      <td>Function to allow customization of the output string.
+        <pre>function(result) {
+    return result.join("");
+}</pre>
+        The <code>result</code> parameter:
+        <ul>
+          <li>Will only contain unique values.</li>
+          <li>If a result filter has been defined in the query (e.g.
+            <code>equivalents.raw.[0,1]</code>), then the <code>result</code>
+            array will only contain those specific results, i.e. the filter
+            will already be applied to the results.
+          </li>
+          <li>If the returned value is an array, it will be automatically be
+            combined into a <code>string</code> using the <code>joiner</code>
+            option. Return a string if you want to combine the array in a
+            different manner.
+          </li>
+        </ul>
+      </td>
+    </tr>
+  </tbody>
+</table>
+
+**Examples**
+
+```js
+const diacritics = require("diacritics-transliterator");
+
+const string = diacritics.replacePlaceholder("'u' is the base for '<!-- diacritics: base=u;equivalents.html_hex -->'", {
+    placeholder: "<!-- diacritics: {data} -->"
+    each: function(data, target) {
+        // choose specific data to return
+        return data[target];
+    },
+    // used to .join() the result if the done function is not defined
+    joiner: ", ",
+    done: function(placeholder, result) {
+        // modify result as desired
+        return result.join(", ");
+    }
+});
+/* string => "'u' is the base for '&#x00FC;, u&#x0308;, &#x00FA;, u&#x0301;'" */
+```
+
+To convert escaped unicode (e.g. `\\u00FC`) into actual unicode, use the [`diacritics.formatUnicode`](#formatunicode) function, or use the `base.raw` data type:
+
+```js
+const diacritics = require("diacritics-transliterator");
+
+const string = diacritics.replacePlaceholder("'u' is the base for '<% diacritics: base=u;equivalents.unicode %>'", {
+    done: function(placeholder, result) {
+        return diacritics.formatUnicode(result.join(", "));
+    }
+});
+/* string => "'u' is the base for 'ü, ü, ú, ú'" */
+```
+
+Invalid strings throw an error
+
+```js
+const diacritics = require("diacritics-transliterator");
+
+const invalid = diacritics.replacePlaceholder();
+/* thrown error => "Error: Invalid input string" */
+```
+
+---
+
+# Related
+
+- [diacritics database](https://github.com/diacritics/database) – the database behind node-transliterator.
+- [diacritics API](https://github.com/diacritics/api) – the API for the diacritics database.
+
+# Contribution and License Agreement
+
+If you contribute to this project, you are implicitly allowing your code to be distributed under [this license][license]. You are also implicitly verifying that all code is your original work.
+
+[license]: https://git.io/v1EBe
```

Comment:
Fuck, man, you're a fucking writing machine 😆 I have flown over it and it looks good to me. I'll test how the documentation feels from a user point of view as soon as I migrate the diacritics module with mark.js.

But, the CLI documentation is missing completely. There's no need to document the CLI methods redundant, but at least specify that there's a CLI and how to see the available commands.

And how to test the CLI in development without installing it from npm?